worlds/jakanddaxter/Rules.py

```diff
@@ -1,102 +1,116 @@
-
     return accessible_orbs
 
 
-# TODO - Until we come up with a better progressive system for the traders (that avoids hard-locking if you pay the
-#  wrong ones and can't afford the right ones) just make all the traders locked behind the total amount to pay them all.
-def can_trade(state: CollectionState,
-              player: int,
-              multiworld: MultiWorld,
-              options: JakAndDaxterOptions,
-              required_orbs: int,
-              required_previous_trade: int = None) -> bool:
+def can_reach_orbs_global(state: CollectionState,
+                          player: int,
+                          world: JakAndDaxterWorld,
+                          bundle: int) -> bool:
 
-    if options.enable_orbsanity == EnableOrbsanity.option_per_level:
-        bundle_size = options.level_orbsanity_bundle_size.value
-        return can_trade_orbsanity(state, player, bundle_size, required_orbs, required_previous_trade)
-    elif options.enable_orbsanity == EnableOrbsanity.option_global:
-        bundle_size = options.global_orbsanity_bundle_size.value
-        return can_trade_orbsanity(state, player, bundle_size, required_orbs, required_previous_trade)
-    else:
-        return can_trade_regular(state, player, multiworld, required_orbs, required_previous_trade)
+    recalculate_reachable_orbs(state, player)
+    return state.has("Reachable Orbs", player, world.orb_bundle_size * (bundle + 1))
+
+
+def can_reach_orbs_level(state: CollectionState,
+                         player: int,
+                         world: JakAndDaxterWorld,
+                         level_name: str,
+                         bundle: int) -> bool:
 
+    recalculate_reachable_orbs(state, player)
+    return state.has(f"{level_name} Reachable Orbs", player, world.orb_bundle_size * (bundle + 1))
 
-def can_trade_regular(state: CollectionState,
+
+def can_trade_vanilla(state: CollectionState,
                       player: int,
-                      multiworld: MultiWorld,
                       required_orbs: int,
-                      required_previous_trade: int = None) -> bool:
+                      required_previous_trade: typing.Optional[int] = None) -> bool:
 
-    # We know that Orbsanity is off, so count orbs globally.
-    accessible_orbs = can_reach_orbs_global(state, player, multiworld)
+    recalculate_reachable_orbs(state, player)  # With Orbsanity Off, Reachable Orbs are in fact Tradeable Orbs.
     if required_previous_trade:
         name_of_previous_trade = location_table[Cells.to_ap_id(required_previous_trade)]
-        return (accessible_orbs >= required_orbs
-                and state.can_reach(name_of_previous_trade, "Location", player=player))
+        return (state.has("Reachable Orbs", player, required_orbs)
+                and state.can_reach_location(name_of_previous_trade, player=player))
     else:
-        return accessible_orbs >= required_orbs
+        return state.has("Reachable Orbs", player, required_orbs)
```

Comment:
You could remove the `else` and just return afterwards since the previous line returns (same for the next function)